lib/pbench/test/unit/server/test_cache_manager.py

```diff
@@ -1580,6 +1733,8 @@ def test_find(
         assert tarball
         assert tarball.name == dataset_name
         assert tarball.resource_id == md5
+        monkeypatch.setattr(Dataset, "query", mock_query)
+        monkeypatch.setattr(Metadata, "getvalue", lambda d, k: tarball.tarball_path)
```

Comment:
Looking at the corresponding mock in `lib/pbench/test/unit/server/test_datasets_compare.py`, this `getvalue()` mock looks over-simplified/generalized:  for one thing, it's missing the third, optional parameter (`user`); and, for another, it doesn't do any kind of check on the requested key value (which seems like it could produce "interesting" and mysterious results...).

Reply:
It's not more complicated than needed here.

Reply:
The fact that it has the "wrong" number of parameters but still works is just fascinating.  I so love Python....